### PR TITLE
Sonos - add documentation for select_source

### PR DESCRIPTION
**Pull request by PeteRager**
## Proposed change
The play_media.select_source service exists and the sonos sources are undocumented.  This pull request provides the missing documentation.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [X] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
  - Introduced the ability to select sources for Sonos devices using the `media_player.select_source` service.
  - Added new custom service `sonos.set_sleep_timer` to set a sleep timer on Sonos speakers.
  - Added new custom service `sonos.clear_sleep_timer` to clear the sleep timer on Sonos speakers.

- **Documentation**
  - Updated documentation to include detailed information and examples for the new services: `media_player.select_source`, `sonos.set_sleep_timer`, and `sonos.clear_sleep_timer`.

### CODE REVIEW

File: source/_integrations/sonos.markdown [hunk 1]
```diff
@@ -295,6 +328,15 @@ Sets a timer that will turn off a speaker by tapering the volume down to 0 after
...
 
+### Service `sonos.set_sleep_timer`
+
+Sets a timer that will turn off a speaker by tapering the volume down to 0 after a certain amount of time. Protip: If you set the sleep_time value to 0, then the speaker will immediately start tapering the volume down.
+
+| Service data attribute | Optional | Description |
+| ---------------------- | -------- | ----------- |
+| `entity_id` | yes | String or list of `entity_id`s that will have their timers set.
+| `sleep_time` | no | Integer number of seconds that the speaker should wait until it starts tapering. Cannot exceed 86399 (one day).
+
```
[the remark this answers is not in the document]
Comment (PeteRager) [reply source/_integrations/sonos.markdown]: updated to remove section

File: source/_integrations/sonos.markdown [hunk 2]
```diff
@@ -248,6 +248,39 @@ data:
   enqueue: play
 ```
 
+## Selecting Source
+
+The `media_player.select_source` service may be used to select specific sources on your Sonos.
```
Comment (frenck) [inline source/_integrations/sonos.markdown:253]: This is a service from the `media_player` platform integration and is documented there.

<https://www.home-assistant.io/integrations/media_player#service-media_playerselect_source>

We don't want to duplicate all services across every integration that provides entities for a platform.

**Comment by frenck** [comment]
As per above comment, this is not the right location for this documentation. On the media player documentation, this has been documented already, so going to close this PR here.

Nevertheless, thanks for being willing to contribute 👍 

../Frenck

File: source/_integrations/sonos.markdown (hunk 2 above)
Comment (PeteRager) [reply source/_integrations/sonos.markdown:253]: @frenck I'm trying to document the platform dependent parameters for Sonos.  For Sonos source can be either TV, Line-in or the name of a favorite as this information is not available anywhere.   I copied this from the Roku documentation and changed the Roku platform dependent parameters to the Sonos ones.  

https://www.home-assistant.io/integrations/roku/#source-automation

**Pull request closed without merging**